File: samplejavaapp/src/main/java/com/yoox/samplejavaapp/common/ImageEncoder.java
package com.yoox.samplejavaapp.common;

import android.content.ContentResolver;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.net.Uri;
import android.util.Base64;

import java.io.ByteArrayOutputStream;
import java.io.FileNotFoundException;
import java.io.InputStream;

public class ImageEncoder {

    public static String encode(Uri uri, ContentResolver contentResolver) {
        final InputStream imageStream;
        try {
            imageStream = contentResolver.openInputStream(uri);
        } catch (FileNotFoundException e) {
            return "";
        }
        final Bitmap selectedImage = BitmapFactory.decodeStream(imageStream);
        return encodeImage(selectedImage);
    }

    private static String encodeImage(Bitmap bitmap) {
        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
        bitmap.compress(Bitmap.CompressFormat.JPEG, 60, byteArrayOutputStream);
        byte[] b = byteArrayOutputStream.toByteArray();
        return Base64.encodeToString(b, Base64.DEFAULT);
    }
}

File: samplekotlinapp/src/main/java/com/yoox/samplekotlinapp/common/ImageEncoder.kt
package com.yoox.samplekotlinapp.common

import android.content.ContentResolver
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.net.Uri
import android.util.Base64
import java.io.ByteArrayOutputStream

object ImageEncoder {

    fun encode(uri: Uri, contentResolver: ContentResolver): String? = kotlin.runCatching {
        contentResolver.openInputStream(uri)
    }.map { inputStream ->
        inputStream.use {
            val bitmap = BitmapFactory.decodeStream(it)
            encodeImage(bitmap)
        }
    }.getOrNull()

    private fun encodeImage(bitmap: Bitmap): String {
        ByteArrayOutputStream().use {
            bitmap.compress(Bitmap.CompressFormat.JPEG, 60, it)
            val b = it.toByteArray()
            return Base64.encodeToString(b, Base64.DEFAULT)
        }
    }
}
